features/netsocket/nsapi_dns.cpp

```diff
@@ -243,8 +243,11 @@ static int nsapi_dns_query_multiple(NetworkStack *stack, const char *host,
         }
 
         const uint8_t *response = packet;
-        dns_scan_response(&response, addr, addr_count);
-        break;
+        if (!dns_scan_response(&response, addr, addr_count)) {
```

Comment:
Don't like using "!" for non-boolean values. "== 0" would have been clearer.

Also, you end up querying every DNS server when the destination doesn't exist (like 5.1 did). The point of extra DNS servers is to provide a backup if one goes down, not to get second opinions.

And you return "OK" if no servers respond.

Better fix is:

     if (dns_scan_response(&response, addr, addr_count) > 0) {
          result = NSAPI_ERROR_OK;
     }
    break;